specs/agents/tracing-api-otel.md

```diff
@@ -0,0 +1,235 @@
+## OpenTelemetry API (Tracing)
+
+[OpenTelemetry](https://opentelemetry.io) (OTel in short) provides a vendor-neutral API that allows to capture tracing, logs and metrics data.
+
+Agents may provide a bridge implementation of OpenTelemetry Tracing API following this specification.
+
+Bridging here means that for each OTel span created with the API, a native span/transaction will be created and sent to APM server.
+
+From the perspective of the application code calling the OTel API, the delegation to a native span/transaction is transparent.
+Also, this means that any OTel processors will be bypassed and ignored by the bridge.
+
+### Spans and Transactions
+
+OTel only defines Spans, whereas Elastic APM relies on both Spans and Transactions.
+OTel allows users to provide the _remote context_ when creating a span, which is equivalent to providing a parent to a transaction or span,
+it also allows to provide a (local) parent span.
+
+As a result, when creating Spans through OTel API with a bridge, agents must implement the following algorithm:
+
+```javascript
+// otel_span contains the properties set through the OTel API
+span_or_transaction = null;
+if (otel_span.remote_contex != null) {
+    span_or_transaction = createTransactionWithParent(otel_span.remote_context);
+} else if (otel_span.parent == null) {
+    span_or_transaction = createRootTransaction();
```

Comment:
@SylvainJuge @felixbarny Do we want/need to spec any behavior about how the bridge will interact with existing instrumentation?  

I'm not sure what the situation is in java-land, but here's a few use cases that outline why I'm asking. 

In the Node.js agent we automatically start transactions whenever there's an HTTP(S) request being handled regardless of the framework (express, hapi, fastify), and I've been thinking we might want/need to suppress this default behavior if we're also going to start a transaction due to an oTel span being created.  (otherwise that's two transactions per request being handled)

Similarly, there's a question of what to do when there's an oTel instrumentation package that we also instrument.  The Node.js agent has instrumentation for the `express` web framework, but OTel also has instrumentation for the `express` web framework.  What should happen when the Node.js is instrumenting an express application and the bridge is active?

Reply:
In Java, there is a clear distinction between OTel API (a regular java library) and the OTel instrumentation (also known as OTel instrumentaion agent), thus:
- the OTel bridge only makes the OTel API delegate to our own tracing implementation
- when creating a bridge, there is no such "double instrumentation" as the existing instrumentations are part of the instrumentation agent and are not included in our agent.
- reusing OTel instrumentations is not part of the current proposal, but might be in the future.

If there is no such distinction with API/instrumentation with other platforms, we should then probably clarify this in the spec for which instrumentation have priority. If that would be the case, it also means that we can't distinguish usages of OTel API to instrument custom spans from OTel instrumentation, and that we would just disable our own instrumentations and replace them with OTel ones. This would go way beyond the "create a bridge for the OTel API" goal, thus I hope it's not the case here.